src/test/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayoutTest.java

```diff
@@ -175,15 +171,35 @@ public void testOutputForOptionalErrorValues() throws JoranException, IOExceptio
         log.info(message, new DummyP3Exception());
 
         String timestamp = "\\d{2}-\\d{2}-\\d{4}";
-        String logger = this.getClass().getCanonicalName();
 
         String output = baos.toString();
 
         assertThat(output).containsPattern(
-            timestamp + INFO + logger + ":\\d+: " + message + "\n"
+            timestamp + INFO + getLogger() + message + "\n"
+        );
+        assertThat(output).containsPattern(
+            timestamp + INFO + getLogger() + "\\[P3\\] 0. " + message + "\n"
+        );
+    }
+
+    @Test
+    public void testExtraLogForEmptyCause() throws JoranException, IOException {
+        configLogback(LOGBACK);
+
+        String message = "test exception not found log";
+
+        log.error(message);
+
+        String logger = AbstractLoggingException.class.getCanonicalName();
```

Comment:
shouldn't `getLogger()` be called here as well?

Reply:
this logger is different. the reason main one was extracted to separate function is too many occurrences as per pmd rule. there are still not too many - didn't extract.

can make two extra private variables actually instead function calls

Reply:
may be good. the current `getLogger()` also adds line number?

Reply:
yes, they go together. if you log anything - it must be at certain line. otherwise it would be just another repetitive string literal